tests/linearizability/model.go

```diff
@@ -64,57 +66,85 @@ var etcdModel = porcupine.Model{
 	DescribeOperation: func(in, out interface{}) string {
 		request := in.(etcdRequest)
 		response := out.(etcdResponse)
-		var resp string
 		switch request.op {
 		case Get:
 			if response.err != nil {
-				resp = response.err.Error()
+				return fmt.Sprintf("get(%q) -> %q", request.key, response.err)
 			} else {
-				resp = response.getData
+				return fmt.Sprintf("get(%q) -> %q, rev: %d", request.key, response.getData, response.revision)
 			}
-			return fmt.Sprintf("get(%q) -> %q", request.key, resp)
 		case Put:
 			if response.err != nil {
-				resp = response.err.Error()
+				return fmt.Sprintf("put(%q, %q) -> %s", request.key, request.putData, response.err)
 			} else {
-				resp = "ok"
+				return fmt.Sprintf("put(%q, %q) -> ok, rev: %d", request.key, request.putData, response.revision)
 			}
-			return fmt.Sprintf("put(%q, %q) -> %s", request.key, request.putData, resp)
 		default:
 			return "<invalid>"
 		}
 	},
 }
 
 func step(state EtcdState, request etcdRequest, response etcdResponse) (bool, EtcdState) {
-	if request.key == "" {
-		panic("Invalid request")
-	}
-	if state.Key == "" {
-		state.Key = request.key
+	if state.LastRevision == 0 {
+		return true, initState(request, response)
 	}
 	if state.Key != request.key {
 		panic("Multiple keys not supported")
 	}
 	switch request.op {
 	case Get:
-		if state.Value == response.getData {
-			return true, state
-		}
-		for write := range state.FailedWrites {
-			if write == response.getData {
-				state.Value = response.getData
-				delete(state.FailedWrites, write)
-				return true, state
-			}
-		}
+		return stepGet(state, request, response)
 	case Put:
-		if response.err == nil {
-			state.Value = request.putData
-		} else {
-			state.FailedWrites[request.putData] = struct{}{}
-		}
+		return stepPut(state, request, response)
+	default:
+		panic("Unknown operation")
+	}
+}
+
+func initState(request etcdRequest, response etcdResponse) EtcdState {
+	var value string
+	switch request.op {
+	case Get:
+		value = response.getData
+	case Put:
+		value = request.putData
```

Comment:
Do you need to check the `response.err`?

Reply:
that's a good point. First request can be an error. There is a low probability of this happening in linearization tests as we wait 1 second before disrupting cluster, but model should be correct by itself.

Reply:
It seems this comment isn't resolved.

Reply:
Done